.ci/Jenkinsfile

```diff
@@ -282,31 +289,89 @@ pipeline {
+            skipDefaultCheckout()
           }
-          steps {
+          steps{
             deleteDir()
-            dir("${OPBEANS_REPO}"){
-              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
-                  url: "git@github.com:elastic/${OPBEANS_REPO}.git"
-              // It's required to transform the tag value to the artifact version
-              sh script: ".ci/bump-version.sh ${env.BRANCH_NAME.replaceAll('^v', '')}", label: 'Bump version'
-              // The opbeans pipeline will trigger a release for the master branch
-              gitPush()
-              // The opbeans pipeline will trigger a release for the release tag
-              gitCreateTag(tag: "${env.BRANCH_NAME}")
+            unstash 'source'
+            dir("${BASE_DIR}") {
+              // Bail out if we don't have the right parameters
+              isValidReleaseVer("${params.release_ver}")
+              sh(label: 'Prepare release toolchain', script: '.ci/scripts/prepare-release-containers.sh')
+              // Generate a preview of the CHANGELOG file for visual inspection
+              script{
+                withCredentials([string(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7', variable: 'GITHUB_TOKEN')]) {
+                  def preview = sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview true", returnStdout: true)
+                  input(message: 'Do you accept these proposed entries for the CHANGELOG.asciidoc file?', ok: 'Yes, proceed with writing them.',
+                        parameters: [text(description: 'IMPORTANT: Proposed changes CANNOT be edited here. Abort the build if not acceptable.',
+                        defaultValue: "${preview}", name: 'preview')])
+                  // Re-run without preview flag and rewrite the file
+                  sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview false > ./CHANGELOG.asciidoc", returnStdout: true)
+                }
+              }
+              // Update the EOL table in the upgrading.asciidoc file
+              sh(label: 'Update EOL table', script: ".ci/scripts/generate_eol.sh --version ${params.release_ver} > ./docs/upgrading.asciidoc")
+              sh(label: 'Update package.json', script: ".ci/scripts/update_package_version.sh ${params.release_ver}")
+              withCredentials([usernamePassword(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken',
+                                          passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+                withGitRelease(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken') {
+                  sh(label: 'Tag and push', script: ".ci/scripts/tag_release.sh --version ${params.release_ver}")
+                }
+              }
+              prepareRelease() {
+                sh 'npm publish'
+              }
```

Comment:
When running `npm publish` in a vanilla workspace, will it download all the dependencies?  If so, I'd say it might be a bit slow that the TOTP that has been queried, might be expired

Reply:
No, publish will just take whatever is in the CWD and publish it. It doesn't care about what's checked into git or not, or the state of `node_modules`. It literally just takes the files (respecting `.npmignore`) and bundles them up into a tar-ball which is pushed to npm.